### PR TITLE
Define roles and their responsibilities in a PR

### PR DESCRIPTION
**Pull request by daxian-dbw**
Reviewer and Assignee of a PR should be separate roles.
- Reviewers do code review, and eventually approve the PR
- Assignee is a maintainer, who monitors the progress of the PR, coordinates the review process, and merges PR after it's approved.
When submitting a new PR, the author is encouraged to choose Reviewer(s) and Assignee for the PR. If either Reviewer or Assignee is left blank, maintainers will choose them for the PR.
However, for PowerShell Team members, You are **required** to choose them yourself for the PR.
**Note that**, even if you are not designated as a Reviewer for a PR, you can still feel free to jump in and review the change.

**The PR author is responsible to drive the PR to the Approved state.** The author should ping the reviewer(s) when they don't respond timely enough (or event talk to them if you can). If the review process is making no progress (or too slow), the author can always contact the Assignee to help coordinate the process, for example, if the reviewer(s) don't respond back for too long, the Assignee can review the existing comments and decide if the PR is in good state or a new reviewer should be pulled in.

A template will be added to pull request, to remind the author to choose Reviewer(s) and Assignee for the PR.

### CODE REVIEW

File: .github/CONTRIBUTING.md
Comment (TravisEz13) [inline .github/CONTRIBUTING.md]: spelling nexe should the next

File: .github/CONTRIBUTING.md
Comment (TravisEz13) [inline .github/CONTRIBUTING.md]: `Author, Reviewer, and Assignee of a PR`

File: .github/CONTRIBUTING.md
Comment (TravisEz13) [inline .github/CONTRIBUTING.md]: unneeded comma

File: .github/CONTRIBUTING.md
Comment (TravisEz13) [inline .github/CONTRIBUTING.md]: missing article, should be `the progress`

File: .github/CONTRIBUTING.md
Comment (TravisEz13) [inline .github/CONTRIBUTING.md]: missing article, should be either `The assignee` or `An assignee`

File: .github/CONTRIBUTING.md
Comment (TravisEz13) [inline .github/CONTRIBUTING.md]: missing article on `Author`

File: .github/CONTRIBUTING.md
Comment (TravisEz13) [inline .github/CONTRIBUTING.md]: `will be assigned`  to `shall be assigned`

File: .github/CONTRIBUTING.md
Comment (TravisEz13) [inline .github/CONTRIBUTING.md]: `to drive` should be `for driving`

File: .github/CONTRIBUTING.md
Comment (TravisEz13) [inline .github/CONTRIBUTING.md]: `will choose` to `shall choose`

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
I found no real conceptual issues.  Just many grammar issues.  Not sure I commented on all of them...

**Comment by daxian-dbw** [comment]
@TravisEz13 thanks for catching them. I think I fixed them all this time, can you take a look again?

File: .github/CONTRIBUTING.md
Comment (TravisEz13) [inline .github/CONTRIBUTING.md]: comma at end of line 175 is not needed

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (TravisEz13) [inline .github/PULL_REQUEST_TEMPLATE.md]: `If you are not a team member`

Replace `Or leave` with `Alternatively, `

After `Or` Replace the first `Them` with `the fields`.

**Review by TravisEz13 — APPROVED** [review]

File: .github/CONTRIBUTING.md
Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: If I remove that comma, that line will change from
> reviews the change of a PR, leaves comments

to

> reviews the change of a PR leaves comments

That doesn't seem right, does it?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (daxian-dbw) [inline .github/PULL_REQUEST_TEMPLATE.md]: Much better. Will fix it.

File: .github/CONTRIBUTING.md
Comment (TravisEz13) [inline .github/CONTRIBUTING.md]: Oh didn't catch that you were missing an `And`.

Comment (TravisEz13) [inline .github/CONTRIBUTING.md]: discussed offline and it looks good .

File: .github/CONTRIBUTING.md
Comment (mirichmo) [inline .github/CONTRIBUTING.md]: Can you make this heading a little more descriptive? Something like, "Roles of a PR: Author, Reviewer, and Assignee"

File: .github/CONTRIBUTING.md
Comment (mirichmo) [inline .github/CONTRIBUTING.md]: This is a little misleading to me. "collaborators" are external people who have special status with the project (Charles, Ilya, Jeff, etc.). Do you mean that anyone can be a reviewer?

On a related note, should we specify that at least one PowerShell team member or a Collaborator must sign off for a review to be Approved?

**Review by mirichmo — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md
Comment (TravisEz13) [inline .github/CONTRIBUTING.md]: Contributor is the more generic term

Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: When clicking the "Reviewer" field, you can select from a list of people, that list is collaborators, right?
I think the author has the freedom to choose anyone from them. It's Assignee's duty to decide if the review is sufficient. If not, the assignee can add more reviewers.

File: .github/CONTRIBUTING.md
Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: That is better.

**Comment by daxian-dbw** [comment]
@mirichmo @TravisEz13 the conversation about "collaborators" got hidden by Github, but that comment still stands.

File: .github/CONTRIBUTING.md
Comment (vors) [inline .github/CONTRIBUTING.md]: The Author

File: .github/CONTRIBUTING.md
Comment (vors) [inline .github/CONTRIBUTING.md]: Reviewer can choose to start review before the successful test pass. I think we should re-phrase it to emphasize that reviwer can postpone reviewer, if tests are failing, but can start earlier.

File: .github/CONTRIBUTING.md
Comment (vors) [inline .github/CONTRIBUTING.md]: Should we mention labels that indicates who is currently responsible for the next step?

**Review by vors — APPROVED** [review]
Tentatively approved

File: .github/CONTRIBUTING.md
Comment (andschwa) [inline .github/CONTRIBUTING.md]: Especially with our flaky tests.

**Review by andyleejordan — APPROVED** [review]
I approve the content of these changes, but agree with Sergei's comments. This is excellent clarification I think.

File: .github/CONTRIBUTING.md
Comment (SteveL-MSFT) [inline .github/CONTRIBUTING.md]: Title should be:

Roles and Responsibilities of a PR: Author, Reviewer, and Assignee

File: .github/CONTRIBUTING.md
Comment (SteveL-MSFT) [inline .github/CONTRIBUTING.md]: A Reviewer can be anyone who wants to contribute

File: .github/CONTRIBUTING.md
Comment (SteveL-MSFT) [inline .github/CONTRIBUTING.md]: Maintainer should be caps, add link to https://github.com/PowerShell/PowerShell/tree/master/docs/maintainers

File: .github/CONTRIBUTING.md
Comment (SteveL-MSFT) [inline .github/CONTRIBUTING.md]: "after it's been approved"

File: .github/CONTRIBUTING.md
Comment (SteveL-MSFT) [inline .github/CONTRIBUTING.md]: Probably also mention that PRs with `Needs Review` label needs a Reviewer

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md
Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: Fixed.

File: .github/CONTRIBUTING.md
Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: Good point. Fixed.

File: .github/CONTRIBUTING.md
Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: I think labels are more useful to maintainers and less useful to reviewers because a reviewer probably will not look at the PR list very often. So I think it's better that the author pings the reviewers when she/he is ready to start the next review iteration.

File: .github/CONTRIBUTING.md
Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: Fixed.

File: .github/CONTRIBUTING.md
Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: Changed to `A Reviewer can be anyone who wants to contribute`.

> Probably also mention that PRs with Needs Review label needs a Reviewer

Do you mean the `Review - Needed` label? I'm afraid this label doesn't mean a reviewer needs to be assigned to a PR. I explicitly requested reviewers in #3170, but the `Review - Needed` label still got added to that PR.

File: .github/CONTRIBUTING.md
Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: Fixed.

File: .github/CONTRIBUTING.md
Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: Fixed.

File: .github/CONTRIBUTING.md
Comment (SteveL-MSFT) [inline .github/CONTRIBUTING.md]: Yes, I meant `Review - Needed`.  I was told (forgot by whom) that was what this label meant.  If not, we should document what exactly this label means (the current docs just has a link another doc without really defining this label)

File: .github/CONTRIBUTING.md
Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: > at least one PowerShell team member or a Collaborator must sign off for a review to be Approved

This information is added. Please take another look.

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by daxian-dbw** [comment]
Hi Reviewers, I think I have addressed all the comments, can you please take another look?

**Review by mirichmo — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (joeyaiello) [inline .github/PULL_REQUEST_TEMPLATE.md]: I would switch this sentence around as the "familiar" path is less likely. So something like:

> If you are not part of the PowerShell Team, you can leave the fields blank and the Maintainers will choose them for you. If you are familiar with the team, feel free to add some Reviewers yourself.

File: .github/CONTRIBUTING.md
Comment (joeyaiello) [inline .github/CONTRIBUTING.md]: > and *an* Assignee

File: .github/CONTRIBUTING.md
Comment (joeyaiello) [inline .github/CONTRIBUTING.md]: > one of *the* Maintainers...

File: .github/CONTRIBUTING.md
Comment (joeyaiello) [inline .github/CONTRIBUTING.md]: > is chosen, *the* Assignee

File: .github/CONTRIBUTING.md
Comment (joeyaiello) [inline .github/CONTRIBUTING.md]: > to choose Reviewer(s) and *an* Assignee *for* the PR

File: .github/CONTRIBUTING.md
Comment (joeyaiello) [inline .github/CONTRIBUTING.md]: >by *the* Assignee as appropriate

**Review by joeyaiello — COMMENTED** [review]
Not a huge deal if these tweaks get made before merge, but if you have the chance, it'd be awesome to fix them.

Overall, this looks awesome and provides some much needed clarity with the process.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (daxian-dbw) [inline .github/PULL_REQUEST_TEMPLATE.md]: Fixed.

File: .github/CONTRIBUTING.md
Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: Fixed.

File: .github/CONTRIBUTING.md
Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: Fixed.

File: .github/CONTRIBUTING.md
Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: Fixed.

File: .github/CONTRIBUTING.md
Comment (daxian-dbw) [inline .github/CONTRIBUTING.md]: Fixed.

**Comment by daxian-dbw** [comment]
@joeyaiello Thanks for the review! I have addressed your comments. Please take another look.

**Review by joeyaiello — APPROVED** [review]
LGTM!

File: .github/CONTRIBUTING.md
Comment (SteveL-MSFT) [inline .github/CONTRIBUTING.md]: Collaborator should have a capital 'C' as it's a specific group

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Comment by iSazonov** [comment]
I can not to choose Reviewer(s) and Assignee for my PR 😕

**Review by lzybkr — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@iSazonov hmm, I didn't know about that ... so I guess then all non-powershell-team users cannot choose Reviewer and Assignee? @TravisEz13 do you happen to know if that's the case?

@SteveL-MSFT, the `collaborator` has been changed to `Collaborator`.

**Comment by TravisEz13** [comment]
@daxian-dbw I just submitted a review to PSScriptAnalyzer and I cannot even assign a reviewer there.

**Comment by iSazonov** [comment]
@daxian-dbw Yes, it is. Personally for me it is not a problem. I can always make a ping by "/cc".

**Pull request merged**
